apps/nextra/pages/en/build/smart-contracts/aptos-standards/fungible-asset.mdx

```diff
@@ -232,7 +232,69 @@ struct Frozen has drop, store {
+
+### Implement the hook function 
+
+To implement a custom hook function, build a module that contains functions with following signature:
+
+```move filename="example"
+public fun withdraw<T: key>(
+    store: Object<T>,
+    amount: u64,
+    transfer_ref: &TransferRef,
+): FungibleAsset {
+    // Your own logic here
+}
+
+public fun deposit<T: key>(
+    store: Object<T>,
+    fa: FungibleAsset,
+    transfer_ref: &TransferRef,
+) {
+    // Your own logic here
+}
+```
+
+There's some limitation in what you will be able to invoke inside of those withdraw/deposit function body due to re-entrancy prevention:
+- When performing the actual withdraw/deposit operations, ONLY call `with_ref` APIs in your custom hooks for deposit/withdraw operations.
+    -  use `fungible_asset::deposit_with_ref` instead of `fungible_asset::deposit` to consume the passed in `FungibleAsset`.
+    -  use `fungible_asset::withdraw_with_ref` instead of `fungible_asset::withdraw` to generate the `FungibleAsset` for customize withdraw operation.
+- Avoid calling into any functions defined in `dispatchable_fungible_asset` and `primary_fungible_store`.
+    - It is ok to call inline functions defined in those two modules.
+    - Calling other functions in those two modules could result in errors upon transfer.
+
+More details of why these limitation is necessary could be found in the [AIP](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-73.md).
+
+### Register the hook function
+
+Once you have the functions implemented, use the api defined in `dispatchable_fungible_asset::register_dispatch_functions` to bind the function with your fungible asset.
+
+```move filename="register_dispatch"
+public fun register_dispatch_functions(
+    constructor_ref: &ConstructorRef,
+    withdraw_function: Option<FunctionInfo>,
+    deposit_function: Option<FunctionInfo>,
+    derived_balance_function: Option<FunctionInfo>,
+)
+```
+
+Note that this `register_dispatch_functions` takes `Option<FunctionInfo>` as input. This is because we can specify for withdraw/deposit/balance operation if we want to use the override logic or the vanilla logic. When `option::none()` is passed in, this fungible asset would just default to the vanilla withdraw/deposit/balance logic. The `constructoor_ref` is the reference for the metadata object of your fungible asset.
+
+Function info can be constructed using the following api:
+
+```move filename="new_function_info"
+public fun new_function_info(module_signer: &signer, module_name: String, function_name: String): FunctionInfo
+```
+
+For security reason you will need to pass in the signer of the module in order to create a `FunctionInfo` refering to functions in this module. Pass in the `module_name` and `function_name` of the withdraw/deposit implementation function 
```

Comment:
````suggestion
Function info can be constructed using the following function:

```move filename="new_function_info"
module 0x1::dispatchable_fungible_asset {
    public fun new_function_info(module_signer: &signer, module_name: String, function_name: String): FunctionInfo
}
```

For security, you will need to pass in the signer of the module to create the `FunctionInfo` functions in the module.  If using the previous example you would call:

```move 
module 0x42::my_fungible_asset {
    use aptos_framework::string;
    use aptos_framework::object;
    use aptos_framework::primary_fungible_store;
    use aptos_framework::dispatchable_fungible_asset;

    fun create_fungible_asset(module_signer: &signer, /* ... */) {
        // Make the deposit override function info
        let deposit_override = dispatchable_fungible_asset::new_function_info(
            module_signer, 
            string::utf8(b"example"), 
            string::utf8("deposit")
        );
        
        // Create the fungible asset
        let constructor_ref = object::create_sticky_object( /* ... */);
  
  
  
  primary_fungible_store::create_primary_store_enabled_fungible_asset(&constructor_ref, ...);
        // or below if not having primary stores
        // fungible_asset::add_fungibility(&constructor_ref, /* ... */);

        // Override default functionality for deposit
        dispatchable_fungible_asset::register_dispatch_functions(
            &constructor_ref,
            option::none(),
            option::some(deposit_override),
            option::none()
        );

        // ...
    }

    // ...
}
```
````